addons/mail/static/src/core/common/composer.js

```diff
@@ -736,10 +739,44 @@ export class Composer extends Component {
 
     saveContent() {
         const composer = toRaw(this.props.composer);
-        const fullComposerContent =
-            document
-                .querySelector(".o_mail_composer_form_view .note-editable")
-                ?.innerText.replace(/(\t|\n)+/g, "\n") ?? composer.text;
+        const editable = document.querySelector(".o_mail_composer_form_view .note-editable");
+        let fullComposerContent;
+        if (editable) {
+            /**
+             * Converts the given HTML element to text
+             * @param {HTMLElement} element
+             * @returns {string}
+             */
+            const htmlToText = element => {
+                let text = "";
+                for (const child of element.childNodes) {
+                    if (child.nodeType === Node.TEXT_NODE) {
+                        text += child.textContent;
+                    } else if (child.nodeType === Node.ELEMENT_NODE) {
+                        if (child.classList.contains("email-signature")) {
+                            continue; // Ignore the email signature
+                        }
+                        if (child.nodeName === "UL") {
+                            for (const li of child.childNodes) {
+                                text += `\t- ${li.textContent}\n`;
+                            }
+                        } else if (child.nodeName === "OL") {
+                            let index = 1;
+                            for (const li of child.childNodes) {
+                                text += `\t${index}. ${li.textContent}\n`;
+                                index++;
+                            }
+                        } else {
+                            text += `${htmlToText(child)}\n`;
+                        }
+                    }
+                }
+                return text;
+            };
```

Comment:
I still think this is kinda lot of work and complexity. I remember that you've shown to me that when you have actual list (OL/UL) done in the chatter like a markdown text, it is translated when switching to the full composer. Have you found where this is done and see if you could use their method to suit your needs here or add to it?